src/test/xmeta/test_symbol_table.cpp

```diff
@@ -517,3 +515,377 @@ TEST_CASE("Resolving types across namespaces test")
+    auto const& it = namespaces.find("N");
+    REQUIRE(it != namespaces.end());
+    auto const& ns_bodies = it->second->get_namespace_bodies();
+    REQUIRE(ns_bodies.size() == 1);
+    auto const& interfaces = ns_bodies[0]->get_interfaces();
+    REQUIRE(interfaces.size() == 4);
+
+    auto const& combo = interfaces.at("IComboBox");
+    auto const& combo_bases = combo->get_all_interface_bases();
+    REQUIRE(combo_bases.size() == 3);
+    REQUIRE(combo_bases.find(interfaces.at("ITextBox")) != combo_bases.end());
+    REQUIRE(combo_bases.find(interfaces.at("IListBox")) != combo_bases.end());
+    REQUIRE(combo_bases.find(interfaces.at("IControl")) != combo_bases.end());
+    REQUIRE(combo->get_all_interface_bases().size() == 3);
+}
+
+TEST_CASE("Interface methods test")
+{
+    std::istringstream test_idl{ R"(
+        namespace N
+        {
+            interface IControl
+            {
+                void Paint();
+                Int32 Draw(Int32 i, Int32 d);
+            }
+        }
+    )" };
+
+    xmeta_idl_reader reader{ "" };
+    reader.read(test_idl);
+    REQUIRE(reader.get_num_syntax_errors() == 0);
+
+    auto const& namespaces = reader.get_namespaces();
+    auto const& it = namespaces.find("N");
+    REQUIRE(it != namespaces.end());
+    auto const& ns_bodies = it->second->get_namespace_bodies();
+    REQUIRE(ns_bodies.size() == 1);
+    auto const& interfaces = ns_bodies[0]->get_interfaces();
+    REQUIRE(interfaces.size() == 1);
+
+    REQUIRE(interfaces.find("IControl") != interfaces.end());
+    auto const& model = interfaces.at("IControl");
+
+    auto const& methods = model->get_methods();
+    REQUIRE(methods.size() == 2);
+
+    auto const& method0 = methods[0];
+    REQUIRE(method0->get_id() == "Paint");
+    REQUIRE(method0->get_return_type() == std::nullopt);
+    REQUIRE(method0->get_formal_parameters().size() == 0);
+
+    auto const& method1 = methods[1];
+    REQUIRE(method1->get_id() == "Draw");
+    auto const& method1_return_type = method1->get_return_type()->get_semantic();
+    REQUIRE((method1_return_type.is_resolved() && std::get<simple_type>(method1_return_type.get_resolved_target()) == simple_type::Int32));
+    auto const& method1_formal_parameters = method1->get_formal_parameters();
+    REQUIRE(method1_formal_parameters.size() == 2);
+    REQUIRE(method1_formal_parameters[0].get_id() == "i");
+    REQUIRE(std::get<simple_type>(method1_formal_parameters[0].get_type().get_semantic().get_resolved_target()) == simple_type::Int32);
+    REQUIRE(method1_formal_parameters[1].get_id() == "d");
+    REQUIRE(std::get<simple_type>(method1_formal_parameters[1].get_type().get_semantic().get_resolved_target()) == simple_type::Int32);
+}
+
+TEST_CASE("Resolving interface method type ref test")
+{
+    std::istringstream test_idl{ R"(
+        namespace N
+        {
+            struct S1
+            {
+            };
+
+            interface IControl
+            {
+                E1 Draw(S1 p1, M.S2 p2);
```

Comment:
Nice.

Reply:
I'd like to see some test coverage for rejecting things like unresolved types for: bases, return types, parameter types.